pilz_control/spec.dox

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2020 Pilz GmbH & Co. KG
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @page Spec
+ *
+ *  @tableofcontents
+ *  @section system_specification System Specification
+ *  @brief System Specification
+ *
+ *  @subsection ISO-10218-1
+ *
+ *
+ *
+ * @subsubsection controller_hold Controller Hold
+ *
+ * @definereq{no_execution_during_hold}
```

Comment:
I would omit the unit-tests from the docu. That is why I had to remove the referencing, too.

Reply:
> I would omit the unit-tests from the docu. That is why I had to remove the referencing, too.

I only reluctantly accept this because our "inability" to do it right, should not lead to the effect of destroying the main purpose of the specification: Giving inside why a test exists and showing if a test is still relevant or not, by showing the corresponding requirement.

Reply:
You could see it like follows: The requirement actually is captured in the test documentation. What is missing at the moment is an automated build routine, that generates the links and uploads the specification to a place where the whole team can read it.
I would say that this should not lead to the effect of polluting the code documentation on RosWiki.